### PR TITLE
Add a new post

### PR DESCRIPTION
**Pull request by queenypingcap**

### CODE REVIEW

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+In this 5-minute tutorial, we will show you how to spin up a standard TiDB cluster using Docker Compose on your local computer, so you can get a taste of its hybrid power, before using it for work or your own project in production. A standard TiDB cluster includes TiDB (MySQL compatible stateless SQL layer), [TiKV](http://bit.ly/tikv_repo_publication) (a distributed transactional key-value store where the data is stored), and [TiSpark](https://github.com/pingcap/tispark) (an Apache Spark plug-in that powers complex analytical queries within the TiDB ecosystem). 
+
+> NOTE: This guide is for Mac users; we will release a guide for Linux and Windows users soon!
+
+Ready? Let’s get started!
+
+# Setting Up:
+
+Before we start deploying TiDB, we’ll need a few things first: `Brew`, `wget`, Docker, and MySQL. If you don’t have them installed already, here are the instructions to get them.
```
Comment (gregwebs) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: lowercase `brew` here and below

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+2. To install `wget`, use the command below in your Terminal: 
+
+    ```bash
+    brew install wget --with-libressl
+
+    ```
+
+3. Install Docker: [https://www.docker.com/community-edition](https://www.docker.com/community-edition).
+
+4. Install MySQL: [https://dev.mysql.com/downloads/mysql/](https://dev.mysql.com/downloads/mysql/).
```
Comment (gregwebs) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Should this be `brew install mysql-client`? I am not using a mac, perhaps a mac user can verify.

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md (hunk 1 above)
Comment (gregwebs) [reply how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: MySQL -> a MySQL client

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    * Go to `Home` and click on the pull down menu to see dashboards of different TiDB components: TiDB, TiKV, PD, entire cluster.
+
+    * You will see a dashboard full of panels and stats on your current TiDB cluster. Feel free to play around in Grafana, e.g. `TiDB-Cluster-TiKV`, or `TiDB-Cluster-PD`.
+
+![Grafana display of TiKV metrics](media/grafana_display_of_tikv_metrics.png)
+
+<center> Grafana display of TiKV metrics </center>
+
+* Now go to TiDB-vision at [http://localhost:8010](http://localhost:8010) (TiDB-vision is a cluster visualization tool we built in-house to see data transfer and load-balancing inside your cluster).
```
Comment (gregwebs) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: is "we built in-house" useful here?

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+    * It looks something like this:
+
+![TiDB-vision](media/tidb_vision.png)
+
+<center> TiDB-vision </center>
+
+# Test TiDB Compatibility with MySQL
+
+As we mentioned, TiDB is MySQL compatible, which you can essentially use TiDB as MySQL slaves with instant horizontal scalability. That’s how many innovative tech companies, like [Mobike](https://www.pingcap.com/blog/Use-Case-TiDB-in-Mobike/), use TiDB.
```
Comment (gregwebs) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: As we mentioned, TiDB is MySQL compatible. You can use TiDB as MySQL...

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+    ```
+
+3. Install Docker: [https://www.docker.com/community-edition](https://www.docker.com/community-edition).
+
+4. Install MySQL: [https://dev.mysql.com/downloads/mysql/](https://dev.mysql.com/downloads/mysql/).
+
+# Spin up a TiDB cluster
+
+Assuming you’ve installed Docker on your laptop without a hitch, let’s deploy TiDB.
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: What if they had a hitch? We should either point them to a troubleshooting document or leave this out ("Now that Docker is set up, let's deploy TiDB!")

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+1. Clone TiDB Docker Compose onto your laptop:
+
+	```bash
+	git clone https://github.com/pingcap/tidb-docker-compose
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Assumes the user has `git` installed. I don't think it's there by default on Macs, is it? Maybe we should tell them to install it in the `brew install` command above?

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+4. Install MySQL: [https://dev.mysql.com/downloads/mysql/](https://dev.mysql.com/downloads/mysql/).
+
+# Spin up a TiDB cluster
+
+Assuming you’ve installed Docker on your laptop without a hitch, let’s deploy TiDB.
+
+1. Clone TiDB Docker Compose onto your laptop:
+
+	```bash
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Indentation is not consistent.

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+2. Change your directory to `tidb-docker-compose`:
+
+    ```bash
+    cd tidb-docker-compose
+    ```
+
+3. Deploy TiDB on your laptop:
+
+    ```bash
+	docker-compose up -d
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Indentation is not consistent.

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+3. Change your directory to `tispark-sample-data`:
+
+	```bash
+	cd tispark-sample-data
+	```
+
+4. Inject the sample test data from sample data folder to MySQL:
+	
+	```sql
+	mysql --local-infile=1 -u root -h 127.0.0.1 -P 4000 < dss.ddl
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: I think we can just remove step 3 and tell them to do `mysql --local-infile=1 -u root -h 127.0.0.1 -P 4000 < tispark-sample-data/dss.ddl`

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+
+3. Change your directory to `tispark-sample-data`:
+
+	```bash
+	cd tispark-sample-data
+	```
+
+4. Inject the sample test data from sample data folder to MySQL:
+	
+	```sql
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: This is bash. :)

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+	use tpch_001;
+	show tables;
+	```
+
+	**Result:** You can see all the tables in `TPCH_001`, like `NATION`, `ORDERS`, etc.
+
+6. Let’s see what’s in the `NATION` table:
+
+    ```sql
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Indentation is not consistent.

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+	ti.tidbMapDatabase("TPCH_001")
+	```
+
+	It looks something like this:
+
+	![Bind TiSpark to this Spark instance](media/bind_tispark_to_this_spark_instance.png)
+
+4. Now, let’s see what’s in the `NATION` table (should be the same as what we saw on our MySQL client):
+
+	```sql
```
Comment (Hoverbear) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: This should be `java` I think.

**Comment by queenypingcap** [comment]
Hi @gregwebs and @Hoverbear , thank you both for the comments, but this PR is just for the preview. I will incorporate all your comments in another branch and the updates will be reflected on https://pingcap.com/blog/how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark/

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md (hunk 2 above)
Comment (queenypingcap) [reply how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: I will try to verify this on my Mac on Monday :)

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+1. To install `Brew`, go [here](https://brew.sh/) .
+
+2. To install `wget`, use the command below in your Terminal: 
+
+    ```bash
+    brew install wget --with-libressl
+
+    ```
+
+3. Install Docker: [https://www.docker.com/community-edition](https://www.docker.com/community-edition).
```
Comment (lysu) [inline how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Besides docker, we aslo need install [Docker Compose](https://docs.docker.com/compose/install/)

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md (hunk 9 above)
Comment (lysu) [reply how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: `mysql --local-infile=1 -u root -h 127.0.0.1 -P 4000 < tispark-sample-data/dss.ddl` meet error `ERROR 2 (HY000) at line 75: File 'customer.tbl' not found (Errcode: 2 - No such file or directory)
`
and I use 
```shell
cd tispark-sample-data
./sample_data.sh
```

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md (hunk 13 above)
Comment (queenypingcap) [reply how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Currently, this demo is for Mac users only. And on Mac, Docker for Mac and Docker Toolbox already include Compose along with other Docker apps.

File: how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md (hunk 9 above)
Comment (Hoverbear) [reply how_to_spin_up_an_htap_database_in_5_minutes_with_tidb_tispark.md]: Ok! :)

**Pull request closed without merging**